cmd/apichecker/apicheck_test.go

```diff
@@ -0,0 +1,658 @@
+		t.Error(err)
+	}
+}
+
+func TestCreateNewCheck(t *testing.T) {
+	t.Parallel()
+	if !canTestTrello() {
+		t.Skip()
+	}
+	err := trelloCreateNewCheck("Gemini")
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestUpdate(t *testing.T) {
+	t.Parallel()
+	var exchCheck, updatedExch HTMLScrapingData
+	for x := range configData.Exchanges {
+		if configData.Exchanges[x].Name == "Exmo" {
+			exchCheck = *configData.Exchanges[x].Data.HTMLData
+		}
+	}
+	info := ExchangeInfo{Name: "Exmo",
+		CheckType: "HTML String Check",
+		Data: &CheckData{HTMLData: &HTMLScrapingData{RegExp: `Last updated on [\s\S]*, 20\d{2}`,
+			Path: "https://exmo.com/en/api/"},
+		},
+	}
+	updatedExchs := update("Exmo", configData.Exchanges, info)
+	for y := range updatedExchs {
+		if updatedExchs[y].Name == "Exmo" {
+			updatedExch = *updatedExchs[y].Data.HTMLData
+		}
+	}
+	if updatedExch == exchCheck {
+		t.Fatal("update failed")
+	}
+}
+
+func TestCheckMissingExchanges(t *testing.T) {
+	t.Parallel()
+	a := checkMissingExchanges()
+	if len(a) > len(exchange.Exchanges) {
+		t.Fatal("invalid response")
+	}
+}
+
+func TestGetChecklistItems(t *testing.T) {
+	t.Parallel()
+	if !canTestTrello() {
+		t.Skip()
+	}
+	_, err := trelloGetChecklistItems()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestUpdateCheckItem(t *testing.T) {
+	t.Parallel()
+	if !canTestTrello() {
+		t.Skip()
+	}
+	err := trelloUpdateCheckItem(trelloListID, "Gemini 1", "incomplete")
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestNameUpdates(t *testing.T) {
+	t.Parallel()
+	tester := []struct {
+		Name          string
+		Status        string
+		ExpectedName  string
+		ErrorExpected bool
+	}{
+		{
+			Name:          "incorrectname",
+			Status:        "incomplete",
+			ErrorExpected: true,
+		},
+		{
+			Name:          "Gemini 2 2",
+			Status:        "incomplete",
+			ErrorExpected: false,
+		},
+		{
+			Name:          " Gemini 23",
+			Status:        "incomplete",
+			ErrorExpected: true,
+		},
+		{
+			Name:          "Gemini 123",
+			Status:        "complete",
+			ExpectedName:  "Gemini 1",
+			ErrorExpected: false,
+		},
+		{
+			Name:          "Gemini",
+			Status:        "complete",
+			ExpectedName:  "Gemini 1",
+			ErrorExpected: false,
+		},
+		{
+			Name:          "Gemini 24 ",
+			Status:        "incomplete",
+			ErrorExpected: false,
+		},
+	}
+	for x := range tester {
+		r, err := nameStateChanges(tester[x].Name, tester[x].Status)
+		if r != tester[x].ExpectedName && err != nil && !tester[x].ErrorExpected {
+			t.Errorf("%d failed, expected %v, %v, got: %v, %v\n", x,
+				tester[x].ExpectedName,
+				tester[x].ErrorExpected,
+				r,
+				err)
+		}
+	}
+}
+
+func TestReadFileData(t *testing.T) {
+	t.Parallel()
+	_, err := readFileData(testJSONFile)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetSha(t *testing.T) {
+	t.Parallel()
+	_, err := getSha("binance-exchange/binance-official-api-docs")
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSetAuthVars(t *testing.T) {
```

Comment:
I'm having difficulties having this test pass when my API keys are set. Clearing them afterwards does not fix it until I also revert the testupdates.json. I don't think running tests should have a permanent effect on test results and testupdates.json shouldn't be saved